src/infrastructure/axios-utils.test.ts

```diff
@@ -0,0 +1,55 @@
+import { AxiosError } from 'axios';
+
+import { withAxiosErrorTranslation } from './axios-utils';
+
+describe('Axios utils', () => {
+	describe('withAxiosErrorTranslation', () => {
+		it('throws the error if it is not an axios error', async () => {
+			const result = withAxiosErrorTranslation(() => {
+				throw new Error('test error');
+			}, 'some context');
+
+			await expect(result).rejects.toThrow('test error');
+		});
+
+		it('can throw a 400 bad request error', async () => {
+			const result = withAxiosErrorTranslation(() => {
```

Comment:
**issue:** @elainefigma , it seems that the test needs additional setup to test error handling for a specific HTTP code. If I understand correctly, currently, all the tests cover only [the default case](https://github.com/atlassian-labs/figma-for-jira/pull/201/files#diff-9b4d89988f79370e3599908d999bfe7e7202a3f22d1d399c42aa863235749915R56).

For example, in order to test `HTTP 400`, you could do the following:

```ts
		it('can throw a 400 bad request error', async () => {
			const result = withAxiosErrorTranslation(() => {
				throw new AxiosError('Bad request', undefined, undefined, undefined, {
					status: 400,
				} as AxiosResponse<unknown, any>);
			});

			await expect(result).rejects.toBeInstanceOf(BadRequestHttpClientError);
		});
```

Alternatively, you could combine similar tests into one and avoid repetitive test setup by using `it.each`:

```
it.each([
			[BadRequestHttpClientError, 'Bad request', 400],
			[UnauthorizedHttpClientError, 'Unauthorized', 401],
			[ForbiddenHttpClientError, 'Forbidden', 403],
			[NotFoundHttpClientError, 'Not found', 404],
		])(
			'should throw a %s when HTTP code is %s',
			async (ErrorType, message, status) => {
				const axiosError = new AxiosError(
					message,
					undefined,
					undefined,
					undefined,
					{
						status,
					} as AxiosResponse<unknown, any>,
				);

				const result = withAxiosErrorTranslation(() => {
					throw axiosError;
				});

				await expect(result).rejects.toBeInstanceOf(ErrorType);
			},
		);
```

Reply:
Ah, excellent catch. Thanks for the code sample! I've updated the tests.